gitlab/mixins.py

```diff
@@ -414,21 +366,16 @@ def set(self, key: str, value: str, **kwargs: Any) -> base.RESTObject:
         server_data = self.gitlab.http_put(path, post_data=data, **kwargs)
         if TYPE_CHECKING:
             assert not isinstance(server_data, requests.Response)
-            assert self._obj_cls is not None
         return self._obj_cls(self, server_data)
 
 
-class DeleteMixin(_RestManagerBase):
-    _computed_path: Optional[str]
-    _from_parent_attrs: Dict[str, Any]
-    _obj_cls: Optional[Type[base.RESTObject]]
-    _parent: Optional[base.RESTObject]
-    _parent_attrs: Dict[str, Any]
-    _path: Optional[str]
-    gitlab: gitlab.Gitlab
-
+class DeleteMixin(base.RESTManager[base.TObjCls]):
     @exc.on_http_error(exc.GitlabDeleteError)
-    def delete(self, id: Optional[Union[str, int]] = None, **kwargs: Any) -> None:
+    def delete(
+        self: base.RESTManager[Any],
+        id: Optional[Union[str, int]] = None,
+        **kwargs: Any,
+    ) -> None:
```

Comment:
```suggestion
    def delete(
        self: base.RESTManager[Any], id: Optional[Union[str, int]] = None, **kwargs: Any
    ) -> None:
```

Related question: do we need the explicit self annotation here? If not, it can probably go back to the one-liner.